entity-store.d.ts

```diff
@@ -13,9 +13,9 @@ type RecordMetadata = {
   version: number;
 };
 
-export type EntityStore = {
-  fetch: (id: string) => Promise<ClassLike>;
-  fetchRecord: (id: string) => Promise<[ClassLike, RecordMetadata]>;
+export type EntityStore<T> = {
```

Comment:
We should also extend this to EntityStoreCreateOptions<T>. I'll do that quickly.